code/game/objects/items/floppy_disk.dm

```diff
@@ -0,0 +1,346 @@
+#define MAX_DISK_STACK_SIZE 10
+#define STACK_PIXEL_STEP 2
+#define MAX_TEXT_LENGTH 30
+
+/obj/item/disk
+	name = "floppy disk"
+	desc = "A generic floppy disk. No way Nanotrasen still uses those, right?"
+	icon = 'icons/obj/devices/floppy_disks.dmi'
+	icon_state = "datadisk3"
+	w_class = WEIGHT_CLASS_TINY
+	inhand_icon_state = "card-id"
+	lefthand_file = 'icons/mob/inhands/equipment/idcards_lefthand.dmi'
+	righthand_file = 'icons/mob/inhands/equipment/idcards_righthand.dmi'
+	drop_sound = 'sound/items/handling/disk_drop.ogg'
+	pickup_sound = 'sound/items/handling/disk_pickup.ogg'
+	unique_reskin = list(
+		"Red" = "datadisk0",
+		"Dark Blue" = "datadisk1",
+		"Yellow" = "datadisk2",
+		"Black" = "datadisk3",
+		"Green" = "datadisk4",
+		"Purple" = "datadisk5",
+		"Grey" = "datadisk6",
+		"Light Blue" = "datadisk7",
+	)
+
+	/// Sticker icons to choose from (as icon states)
+	var/list/icon_variants = list(
+		"One" = "o_one",
+		"Two" = "o_two",
+		"Three" = "o_three",
+		"Four" = "o_four",
+		"Five" = "o_five",
+		"Six" = "o_six",
+		"Seven" = "o_seven",
+		"Eight" = "o_eight",
+		"Nine" = "o_nine",
+		"Zero" = "o_zero",
+		"A" = "o_A",
+		"B" = "o_B",
+		"C" = "o_C",
+		"D" = "o_D",
+		"E" = "o_E",
+		"F" = "o_F",
+		"Code" = "o_code",
+		"DNA — Green" = "o_dna1",
+		"DNA — Red" = "o_dna2",
+		"Medical" = "o_medical",
+		"Holographic" = "o_holo",
+		)
+
+	/// Toggles the readonly state
+	var/read_only = FALSE
+	/// If readonly state can be toggled on and off
+	var/read_only_locked = FALSE
+	/// Was the sticker icon already selected?
+	var/sticker_changed = FALSE
+	/// Custom description
+	var/custom_description
+	/// Initial sticker overlay
+	var/initial_sticker_icon_state
+
+/obj/item/disk/Initialize(mapload)
+	. = ..()
+	if(initial_sticker_icon_state)
+		add_overlay(initial_sticker_icon_state)
+		sticker_changed = TRUE
+	else if(initial_sticker_icon_state == NONE)
+		return
+	else
+		add_overlay("o_empty")
```

Comment:
broadly I feel all this all shouldn't be using `add_overlay(...)` and `cut_overlays()`, but say adding `sticker_icon_state` as the overlay on `update_overlays()`, and making a proc for setting `sticker_icon_state` and calling `update_appearance(UPDATE_OVERLAYS)` and using that where you need to change it
i.e.
```dm
/obj/item/disk/update_overlays()
	. = ..()
	if(sticker_icon_state)
		. += sticker_icon_state
```
or defaulting to `"o_empty"` if nothing else, cause it looks like that's your implied default?
that handles removing it and all the other stuff itself
avoids needing to manage all this overlay juggling like dna disks adding an overlay over the initial overlay on init, or weird stuff like spraypainting probably messing with the current overlay logic